dot/node_test.go

```diff
@@ -29,7 +29,17 @@ import (
 
 func DefaultTestWestendDevConfig(t *testing.T) *cfg.Config {
 	config := westenddev.DefaultConfig()
-	config.BasePath = t.TempDir()
+	var tempDir string
+	if os.Getenv("CI") == "buildjet" {
+		var err error
+		tempDir, err = os.MkdirTemp("..", "DefaultTestWestendDevConfig")
+		if err != nil {
+			t.Error(err)
+		}
+	} else {
+		tempDir = t.TempDir()
+	}
```

Comment:
`t.TempDir()` creates a temporal dir and then it clean up that directory. 
Are we cleaning up the directory created by `os.MkdirTemp`?

Why is `t.TempDir()` not enough?

Reply:
`t.TempDir` cleans up the folder after the test completes.  From what I gather, there's something about the way it cleans up the folders that is deleting files it shouldn't be cleaning up.  **This only happens on buildjet.**  So the way I solved it, was to just not prune the temp directory and place it outside of `/tmp` for CI.

Reply:
Can we encapsulate this logic in a utils function to prevent duplicating code?

Reply:
Since this only happens in the `dot` package I would prefer to keep this "hackery" contained.  Ideally we don't have to do this.  I think it's something to do with not waiting on a goroutine to complete in the `dot` package, which is causing the problem.  Keeping the fix ugly will remind us to revisit at some point.

Reply:
Could you then please add TODO comment describing all this above

Reply:
I have a feeling that problem happens becouse of timeout, since timout kills the testing process without calling testin package pipline that executes testing.Cleanup in the end.